### ISSUE TITLE
Is that possible to base on SpinalHDL/docker ?

### ISSUE THREAD

**Issue by Readon** [zao111222333/spinalhdl-online#5]
The SpinalHDL/docker now contains all elaboration and formal verification tools.
If this project can base on that with Gtk online support then a lot of man power can be saved.

**Comment by Readon** [comment]
Also, that image is release on Github's docker mirror which is free for open source project with unlimited downloads.

**Comment by zao111222333** [comment]
https://github.com/SpinalHDL/docker

**Comment by zao111222333** [comment]
Good idea, it seems that https://github.com/SpinalHDL/docker can keep SpinalHDL and other tools in latest version, I will try to build code-server/nginx/Gtk base on that image.

**Comment by Readon** [comment]
do you have any good solution for git gui? I mean with web view. I have tested the git-graph extension before. however, it fails and could not be fixed.

**Comment by zao111222333** [comment]
> do you have any good solution for git gui? I mean with web view. I have tested the git-graph extension before. however, it fails and could not be fixed.

[`eamodio.gitlens`](https://marketplace.visualstudio.com/items?itemName=eamodio.gitlens) works well on my situation, you can have a try. 

By the way, since there is no SpinalHDL in my normal work, I will work on above tasks this weekend or later.

**Comment by zao111222333** [comment]
There is always existing an issue for `broadwayd` (html back-end of gtk) with docker, making it not decent for official image to use  `gtk-online`. I will keep following this issue.

See more here: https://gitlab.gnome.org/GNOME/gtk/-/issues/6058
